### ISSUE TITLE
[Select] VoiceOver cuts off last two characters of text inside Select component

### ISSUE THREAD

**Issue by pgouldmiro** [radix-ui/primitives#3671]
## Bug report
When the Select component is focused, VoiceOver cuts off the last two letters of the text inside it.

For example, when the text inside the Select component is "Banana", VoiceOver announces it as `Bana, [list box name], list box pop-up collapsed, combo box`.

Further details:
- This only happens on Safari. It works as expected in Chrome and Firefox.
- This is reproducible on the [Radix site](https://www.radix-ui.com/primitives/docs/components/select) without any custom implementation.
- This only happens with the Radix Select component. Native `<select>` elements (e.g. [W3 Schools example](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select)) work as expected.

### Current Behaviour
VoiceOver cuts off the last two letters of the text inside the Select component – e.g. `Bana, [list box name], list box pop-up collapsed, combo box`

https://github.com/user-attachments/assets/37860f11-f0d7-4f53-a22d-7257d230993b

### Expected behaviour
VoiceOver should announce the entire text inside the Select component, as per the native `<select>` element – e.g. `Banana, [list box name], list box pop-up collapsed, combo box`

### Reproducible example
1. Enable VoiceOver (System Settings > Accessibility > VoiceOver > Turn on VoiceOver)
2. In Safari, navigate to a Radix Select component (e.g. on the [Radix site](https://www.radix-ui.com/primitives/docs/components/select))
3. Press `Tab` or `VO + Right arrow` until the Select component is focused
4. Observe the VoiceOver output (either as speech or using the caption panel)

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Browser          | Safari | 18.5 (20621.2.5.11.8) |
| Assistive tech   | VoiceOver | N/A |
| Operating System | MacOS | Sequoia 15.5 (24F74) |
